File: src/main/java/cn/cloudself/model/AppEntranceEntity.java
package cn.cloudself.model;

import javax.persistence.*;

/**
 * @author HerbLuo
 * @version 1.0.0.d
 *          <p>
 *          change logs:
 *          2017/4/13 HerbLuo 首次创建
 */
@Entity
@Table(name = "app_entrance", schema = "shop")
public class AppEntranceEntity {
    private int id;
    private boolean enabled;
    private byte index;
    private String name;
    private String img;
    private String link;

    @Id
    @Column(name = "id", nullable = false)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "enabled", nullable = false)
    public boolean getEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    @Basic
    @Column(name = "index", nullable = false)
    public byte getIndex() {
        return index;
    }

    public void setIndex(byte index) {
        this.index = index;
    }

    @Basic
    @Column(name = "name", nullable = false, length = 6)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Basic
    @Column(name = "img", nullable = false, length = 255)
    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    @Basic
    @Column(name = "link", nullable = false, length = 255)
    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AppEntranceEntity that = (AppEntranceEntity) o;

        if (id != that.id) return false;
        if (enabled != that.enabled) return false;
        if (index != that.index) return false;
        if (name != null ? !name.equals(that.name) : that.name != null) return false;
        if (img != null ? !img.equals(that.img) : that.img != null) return false;
        return link != null ? link.equals(that.link) : that.link == null;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (enabled ? 1 : 0);
        result = 31 * result + (int) index;
        result = 31 * result + (name != null ? name.hashCode() : 0);
        result = 31 * result + (img != null ? img.hashCode() : 0);
        result = 31 * result + (link != null ? link.hashCode() : 0);
        return result;
    }
}

File: src/main/java/cn/cloudself/model/AppEntranceEntity.kt
package cn.cloudself.model

import javax.persistence.*

/**
 * @author HerbLuo
 * @version 1.0.0.d
 *
 *
 * change logs:
 * 2017/4/13 HerbLuo 首次创建
 */
@Entity
@Table(name = "app_entrance", schema = "shop")
data class AppEntranceEntity(
        @get:Id
        @get:Column(name = "id", nullable = false)
        var id: Int = 0,

        @get:Basic
        @get:Column(name = "enabled", nullable = false)
        var enabled: Boolean = false,

        @get:Basic
        @get:Column(name = "index", nullable = false)
        var index: Byte = 0,

        @get:Basic
        @get:Column(name = "name", nullable = false, length = 6)
        var name: String? = null,

        @get:Basic
        @get:Column(name = "img", nullable = false, length = 255)
        var img: String? = null,

        @get:Basic
        @get:Column(name = "link", nullable = false, length = 255)
        var link: String? = null
)